appinventor/blocklyeditor/src/workspace_svg.js

```diff
@@ -589,18 +589,54 @@ Blockly.WorkspaceSvg.prototype.customContextMenu = function(menuOptions) {
   /**
    * Function that returns a name to be used to sort blocks.
    * The general comparator is the block.category attribute.
-   * In the case of 'Components' the comparator is the instanceName of the component if it exists
-   * (it does not exist for generic components).
    * In the case of Procedures the comparator is the NAME(for definitions) or PROCNAME (for calls)
+   * In the case of 'Components' the comparator is the type of the component
    * @param {!Blockly.Block} block the block that will be compared in the sortByCategory function
    * @returns {string} text to be used in the comparison
    */
   function comparisonName(block){
-    if (block.category === 'Component' && block.instanceName)
-      return block.instanceName;
-    if (block.category === 'Procedures')
+    if (block.category == 'Variables'){
+      return block.category;
+    }
+    if (block.category === 'Procedures'){
       return (block.getFieldValue('NAME') || block.getFieldValue('PROCNAME'));
-    return block.category;
+    }
+    // Floating blocks that are not Component
+    return block["type"];
```

Comment:
Use `block.type` here rather than `block["type"]` due to how certain optimizations work when we compile for production.